docs/getting-started-guide.md

```diff
@@ -0,0 +1,168 @@
+# How to set up and run the Co-Simulation Demo
+
+This demonstration shows how to compile and run the Co-Simulation demo of Buildroot in QEMU with a simulated device in SystemC. This configeration is tested working for Ubuntu 18.0.4 and assumes that `cosim` is located in your home directory. This walkthough also assumes the device being emulated is the Zynq-7000 SoC.
+
+## Dependencies
+Below are the dependencies needed to compile all the libraries in this demo:
+```bash
+sudo apt update
+sudo apt install cmake gmake gcc qemu-kvm qemu-system qemu-user-static verilator
+```
+
+## Setup and Compilation
+Run these commands to clone and build the necessary repos (`~/cosim` assumed as the base directory).
+
+### Create the base directory
+```bash
+mkdir ~/cosim
+```
+
+### SystemC Setup
+```bash
+cd ~/cosim
+SYSC_VERSION=systemc-2.3.2
+wget https://www.accellera.org/images/downloads/standards/systemc/systemc-2.3.2.tar.gz
+tar xf ${SYSC_VERSION}.tar.gz && cd ${SYSC_VERSION}/
+
+# Build and install the binaries
+CXXFLAGS=-std=c++11 ./configure ; make
+sudo make install
+```
+
+### QEMU Setup
+```bash
+cd ~/cosim
+git clone https://github.com/Xilinx/qemu.git
+cd qemu
+git checkout 7e3e3ae09af8abe70383a5
+git submodule update --init dtc
+
+# Configure and build
+./configure --target-list="arm-softmmu,aarch64-softmmu,microblazeel-softmmu" --enable-fdt --disable-kvm --disable-xen
+make
+sudo make install
+export SYSTEM=/cosim/systemc-2.3.2/
+```
+
+### Demo Setup
+```bash
+# Clone the repo and the submodules
+cd ~/cosim
+git clone https://github.com/Xilinx/systemctlm-cosim-demo.git
+cd systemctlm-cosim-demo
+sed -i 's~../libsystemctlm-soc.git~https://github.com/Xilinx/libsystemctlm-soc.git~g' .gitmodules
+git submodule sync --recursive && git submodule update --init libsystemctlm-soc
+
+# Create the Makefile configeration
+cat << EOF | tee .config.mk
+CXXFLAGS=-std=c++11
+SYSTEMC_INCLUDE=${HOME}/cosim/systemc-2.3.2/src
+SYSTEMC_LIBDIR=${HOME}/cosim/systemc-2.3.2/src/.libs
+HAVE_VERILOG=n
+HAVE_VERILOG_VERILATOR=n
+HAVE_VERILOG_VCS=n
+VM_TRACE=1
+EOF*
+
+# Build the demo
+make
+```
+
+### Buildroot Setup
+```bash
+cd ~/cosim
+git clone https://github.com/buildroot/buildroot.git
+cd buildroot
+git checkout 36edacce9c2c3b90f9bb11
+mkdir handles
+```
+
+For this walkthough we will be configuring the demo to emulate the Zynq-7000 SoC. The `.dtsi` files available in the Xilinx repositories need significat modification to operate.
```

Comment:
Not sure I would say that they need a lot of mods like that.  Maybe stage the modified files as references with this documentation and then have the build pull those in?  Since the ones you pull in here won't work.

Reply:
Thanks, yes, I agree with the comments. It would be nice if we could avoid carrying a copy of the dtsi files in this project.

The idea with the zynq-pl-remote-port.dtsi file in the qemu-devicetrees is to act as an overlay (or an example of an overlay) for a zynq-7000 dts file. If you could provide a diff of the changes you had to do to the dts file, perhaps we can make those changes in the original repository.

Ideally, here we could just do a bunch of wgets to get hold of dts files and a line compile the final dtb (perhaps with a small modification to get zynq-pl-remoteport.dtsi included).

Reply:
One more thing that may be causing confusion.

QEMU for the Zynq-7000 takes two DTBs as input, -dtb and -hw-dtb.

If only -dtb is passed, that file is used as both the -dtb and -hw-dtb.

-hw-dtb is used by QEMU to instantiate the machine.
-dtb is passed to the ARM Linux kernel.

This means that the -hw-dtb should include the zynq-pl-remoteport.dtsi but the -dtb should not.

On the opposite side, -dtb should include the amba_pl peripherals implemented in the programmable logic (e.g whatever IP you're cosimulating with) so that Linux drivers can bind. The -hw-dtb file should not include those descriptions.

Reply:
I agree, that would be a better way of including those files. We were just attempting to make it as close to plug and play as possible. I can take a look at the diffs and see how substantial they are.

Reply:
@edgarigl Are you referring to the [linux-xlnx repo](https://github.com/Xilinx/linux-xlnx) repo or [device-tree-xlnx](https://github.com/Xilinx/device-tree-xlnx)?

Reply:
Hi,

Ideally, it should work to:
1. wget arch/arm/boot/dts/zynq-zc702.dts from upstream Linux (e.g https://github.com/torvalds/linux.git) or an equivalent dts from the Xilinx repos you mention. Build and use this as -dtb option to QEMU.
2. Remove existing amba_pl nodes from the dts (if any).
3. Add a line at the bottom of the dts so that it includes the zynq-pl-remoteport.dtsi overlay from https://github.com/Xilinx/qemu-devicetrees.git. Use this as the -hw-dtb option to QEMU.

Does that make sense?

Best regards,
Edgar

Reply:
It's a good point with the plug and play though. An option is to create a script that automates the steps. Perhaps also to include the generated dts files in a sub-directory as Matthew suggested. From my side, I'm open to suggestions on what you guys think is the best way.